x/internal/outline/access_key.go

```diff
@@ -42,13 +42,14 @@ func (p Prefix) String() string {
 	return string(runes)
 }
 
-// TODO(fortuna): provide this as a reusable library. Perhaps as x/shadowsocks or x/outline.
-func parseAccessKey(accessKey string) (*sessionConfig, error) {
-	var config sessionConfig
+func ParseAccessKey(accessKey string) (*SessionConfig, error) {
```

Comment:
Note that this function is not ready for production. The old function supports parsing the [legacy Shadowsocks format](https://archive.is/PW7dL) ([code](https://github.com/Jigsaw-Code/outline-shadowsocksconfig/blob/6d01c4b1beae3a3bc2a1074e5e310dfa023be57d/src/shadowsocks_config.ts#L225)).

We will need to make sure it's on par with the existing code. A good way to do that is by processing the examples from https://github.com/Jigsaw-Code/outline-shadowsocksconfig/blob/master/src/shadowsocks_config.spec.ts.

You will have to address that before we can use it in the Outline client.